3rd-party-tools/slide-tags/spatial-count.jl

```diff
@@ -0,0 +1,561 @@
+
+# Create a downsampling curve
+downsampling = UInt32[]
+table = countmap(df.reads)
+for p in 0:0.05:1
+    s = [length(unique(floor.(sample(0:k*v-1, round(Int,k*v*p), replace=false)/k))) for (k,v) in zip(keys(table),values(table))]
+    append!(downsampling, sum(s))
+    GC.gc()
+end
+
+##### Save results #############################################################
+
+print("Saving results... ") ; flush(stdout)
+
+h5open(joinpath(out_path, "SBcounts.h5"), "w") do file
+    create_group(file, "lists")
+    file["lists/cb_list", compress=9] = String31.(cb_whitelist) # Vector{String}
+    file["lists/sb_list", compress=9] = String31.(sb_whitelist) # Vector{String}
+    file["lists/puck_list"] = String255.(basename.(puck_paths)) # Vector{String}
+    file["lists/R1_list"] = String255.(basename.(R1s))          # Vector{String}
+    file["lists/R2_list"] = String255.(basename.(R2s))          # Vector{String}
+
+    create_group(file, "matrix")
+    file["matrix/cb_index", compress=9] = df.cb_i # Vector{UInt32}
+    file["matrix/umi", compress=9] = df.umi_i     # Vector{UInt32}
+    file["matrix/sb_index", compress=9] = df.sb_i # Vector{UInt32}
+    file["matrix/reads", compress=9] = df.reads   # Vector{UInt32}
+
+    create_group(file, "puck")
+    file["puck/sb", compress=9] = String31.(puckdf.sb)      # Vector{String}
+    file["puck/x", compress=9] = puckdf.x                   # Vector{Float64}
+    file["puck/y", compress=9] = puckdf.y                   # Vector{Float64}
+    file["puck/puck_index", compress=9] = puckdf.puck_index # Vector{UInt8}
+    
+    create_group(file, "metadata")
+    
+    file["metadata/switch"] = convert(Int8, switch)
+    file["metadata/bead"] = String7(bead)
+    file["metadata/num_lowQbeads"] = num_lowQbeads
+    
+    file["metadata/reads"] = metadata["reads"]
+    file["metadata/reads_filtered"] = metadata["reads_filtered"]
+    file["metadata/R1_tooshort"] = metadata["R1_tooshort"]
+    file["metadata/R2_tooshort"] = metadata["R2_tooshort"]
+
+    create_group(file, "metadata/UMI")
+    file["metadata/UMI/type"] = keys(metadata["UMI"]) |> x -> String15.(x) |> collect
+    file["metadata/UMI/count"] = values(metadata["UMI"]) |> collect
+    
+    create_group(file, "metadata/UP")
+    file["metadata/UP/type"] = keys(metadata["UP"]) |> x -> String15.(x) |> collect
+    file["metadata/UP/count"] = values(metadata["UP"]) |> collect
+
+    create_group(file, "metadata/SB")
+    file["metadata/SB/type"] = keys(metadata["SB"]) |> x -> String15.(x) |> collect
+    file["metadata/SB/count"] = values(metadata["SB"]) |> collect
+
+    create_group(file, "metadata/SB_HD")
+    file["metadata/SB_HD/type"] = keys(metadata["SB_HD"]) |> collect
+    file["metadata/SB_HD/count"] = values(metadata["SB_HD"]) |> collect
+    
+    file["metadata/downsampling"] = downsampling # Vector{UInt32}
+end;
+
+println("done") ; flush(stdout) ; GC.gc()
+
+##### Documentation ############################################################
```

Comment:
Do we want to keep this in the same code doc or make a .md file?

Reply:
Were not using this code -- I had it for testing. Were pulling the script from their repo. Im going to remove it